### PR TITLE
Sample script flockOfBirds.js with sounds and flocking behavior

### PR DESCRIPTION
**Pull request by PhilipRosedale**
This script will create a number of entities that 'fly' around in a box defined at start of script and play bird 'chirp' sounds.

### CODE REVIEW

File: examples/FlockOfbirds.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+                    birds[i].audioId = Audio.playSound(birds[i].sound, options);
+                }
+                numPlaying++;
+                // Change size 
+                Entities.editEntity(birds[i].entityId, { dimensions: Vec3.multiply(1.5, properties.dimensions)});
+                
+            } else if (birds[i].audioId) {
+                // If bird is playing a chirp 
+                if (!birds[i].audioId.isPlaying) {
+                    // clear ID if playing has stopped
```
Comment (howard-stearns) [inline examples/FlockOfbirds.js]: This comment is no longer applicable.

File: examples/FlockOfbirds.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                var newVelocity = Vec3.sum(randomVelocity, Vec3.sum(alignmentVelocity, cohesionVelocity));
+
+                Entities.editEntity(birds[i].entityId, { velocity: Vec3.sum(properties.velocity, newVelocity) });
+
+            } 
+
+            //  Check whether to play a chirp
+            if (playSounds && (!birds[i].audioId || !birds[i].audioId.isPlaying) && (Math.random() < ((numPlaying > 0) ? SOUND_PROBABILITY / numPlaying : SOUND_PROBABILITY))) {
+                var options = {
+                          position: properties.position,
```
Comment (howard-stearns) [inline examples/FlockOfbirds.js]: Whitespace (not holding merge).

File: examples/FlockOfbirds.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    if (birdVelocitiesCounted > 0) {
+        averageVelocity = Vec3.multiply(1.0 / birdVelocitiesCounted, sumVelocity);
+        //print(Vec3.length(averageVelocity));
+    } 
+    if (birdPositionsCounted > 0) {
+        averagePosition = Vec3.multiply(1.0 / birdPositionsCounted, sumPosition);
+    }
+}
+
+loadBirds(NUM_BIRDS);
```
Comment (ZappoMan) [inline examples/FlockOfbirds.js]: This is going to cause this script to not work in an AC.

File: examples/FlockOfbirds.js [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+                                                            z: lowerCorner.z + (upperCorner.z - lowerCorner.z) / 2.0  },
+                                               dimensions: { x: (upperCorner.x - lowerCorner.x), y: FLOOR_THICKNESS, z: (upperCorner.z - lowerCorner.z)},
+                                               color: {red: 100, green: 100, blue: 100}
+                                });
+}
+// Connect a call back that happens every frame
+Script.update.connect(updateBirds);
+
+//  Delete our little friends if script is stopped
+Script.scriptEnding.connect(function() {
```
Comment (ZappoMan) [inline examples/FlockOfbirds.js]: another issue with running this script in an AC is that we can't guarantee that when the script is shut down, the other assignments haven't previously shut down. So this construct of deleting entities in the "scriptEnding" signal will not always work. Sometimes the birds will get deleted, sometimes they won't.. it all depends on how the domain server stops it's assignments.

File: examples/FlockOfbirds.js (hunk 3 above)
Comment (howard-stearns) [inline examples/FlockOfbirds.js]: What was the AC issue? I didn't follow that..

File: examples/FlockOfbirds.js [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+    for (var i = 0; i < birds.length; i++) {
+        Entities.deleteEntity(birds[i].entityId);
+    }
+    if (floor) {
+        Entities.deleteEntity(floor);
+    }
+});
+
+function loadBirds(howMany) {
+  while (!Entities.serversExist() || !Entities.canRez()) {
```
Comment (ZappoMan) [inline examples/FlockOfbirds.js]: this is not needed here, since you check it in updateBirds() before calling this

**Pull request merged**